### ISSUE TITLE
[Feature Request]: CLIPSeg as a model

### ISSUE THREAD

**Issue by ComplexRobot** [Bing-su/adetailer#369]
### Is your feature request related to a problem? Please describe.

One can only inpaint face/hands/person unless they obtain a dataset and use ultralytics to train a model for something else. Furthermore, a new model must be trained for every new object one might want to mask.

### Describe the solution you'd like

Provide support for [clipseg](https://huggingface.co/docs/transformers/main/en/model_doc/clipseg), a model in the [transformers](https://huggingface.co/docs/transformers/index) library, which generates a segmentation mask from a text prompt. 
It's not always accurate, but the ability to segment something based on a text prompt is a powerful tool. [CLIPSegForImageSegmentation](https://huggingface.co/docs/transformers/main/en/model_doc/clipseg#transformers.CLIPSegForImageSegmentation) takes an arbitrary prompt to produce a segmentation. 👌

### Describe alternatives you've considered

I tried searching online to find pretrained ultralytics models to segment what I want. I found nothing after hours of searching. 😅 I also don't think I could train a model myself. But I found out about an extension called [txt2mask](https://github.com/ThereforeGames/txt2mask) which uses clipseg to produce masks. However, adetailer is still the best extension for doing automatic inpainting. It's just limited by what can be inpainted.

### Additional context

Thank you for the wonderful extension!

**Comment by 4lt3r3go** [comment]
i see what you mean but hey, training a model for adetailer is not that hard it seems, there are a bunch shared here and there already if you havent noticed, try search "adteailer" on civit

**Comment by ComplexRobot** [comment]
> i see what you mean but hey, training a model for adetailer is not that hard it seems, there are a bunch shared here and there already if you havent noticed, try search "adteailer" on civit. However, a precise mask would be really more powerfull, out than the standard box

Thanks for the tip! I wasn't sure what to search for.

**Comment by Bing-su** [comment]
In fact, the code I wrote for training the YOLO model is almost 10 lines long.

To the original question: There are extensions like [sd-webui-segment-anything](https://github.com/continue-revolution/sd-webui-segment-anything).

**Comment by ComplexRobot** [comment]
> In fact, the code I wrote for training the YOLO model is almost 10 lines long.
> 
> To the original question: There are extensions like [sd-webui-segment-anything](https://github.com/continue-revolution/sd-webui-segment-anything).

Well, if it's that easy, I might look into it. I'm mostly concerned with how long it will take to download the training data and if my GPU can handle it well. I guess I could train remotely, I'll have to look into it.

I already have Segment Anything. I'm most interested in something that can do everything automatically, because it saves a lot of time. Adetailer applies the inpainting automatically and I didn't like some other methods I used for that, such as comfy-ui.
